pallets/automation-time/src/lib.rs

```diff
@@ -397,8 +401,16 @@ pub mod pallet {
 			encoded_call: Vec<u8>,
 			encoded_call_weight: Weight,
 			overall_weight: Weight,
+			instruction_sequence: InstructionSequence,
+			schedule_as: Option<T::AccountId>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
+
+			// Make sure the owner is the proxy account of the user account.
+			if let Some(schedule_as_account) = schedule_as.clone() {
+				T::EnsureProxy::ensure_ok(schedule_as_account, who.clone())?;
+			}
```

Comment:
Yeah, this is definitely needed. This will be skipped if schedule_as is null?

Reply:
Yes, this will be skipped if schedule_as is null.